lib/search/embeddings/bumblebee_provider.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Search.Embeddings.BumblebeeProvider do
+  alias Search.{Embeddings, Packages, Repo}
+  import Ecto.Query
+
+  @behaviour Embeddings.Provider
+
+  @impl true
+  def child_spec(opts) do
+    %{serving_name: serving_name} = parse_opts(opts)
+
+    {Nx.Serving, serving: load_model(opts), name: serving_name, batch_size: get_batch_size(opts)}
```

Comment:
Bumblebee sets the batch_size based on the compilation options already :)

```suggestion
    {Nx.Serving, serving: load_model(opts), name: serving_name}
```